kernel/util/mlog-base/util/Logger.hh

```diff
@@ -51,51 +51,79 @@ namespace mlog {
-    void setName(char const* name) { this->name = name; }
+template<class Filter = FilterAny>
+class Logger
+  : public Filter
+{
+public:
+  Logger(char const* name = "mlog") : name(name) {}
+  Logger(char const* name, Filter filter) : Filter(filter), name(name) {}
+  Logger(Filter filter) : Filter(filter), name("mlog") {}
+  void setName(char const* name) { this->name = name; }
 
-    template<class MSG, typename... ARGS>
-    void log(ARGS&&... args) const {
-      if (this->isActive(MSG::VERBOSITY)) write<MSG, ARGS...>(std::forward<ARGS>(args)...);
-    }
+  template<class MSG, typename... ARGS>
+  void log(ARGS&&... args) const {
+    if (this->isActive(MSG::VERBOSITY)) write<MSG, ARGS...>(std::forward<ARGS>(args)...);
+  }
 
-    template<class MSG, typename... ARGS>
-    void write(ARGS&&... args) const NOINLINE;
+  template<class MSG, typename... ARGS>
+  void write(ARGS&&... args) const NOINLINE;
 
-    void flush() { sink->flush(); }
+  void flush() { sink->flush(); }
 
-    // syntactic sugar
-    template<typename... ARGS>
-    void detail(ARGS&&... args) { log<TextDetail>(std::forward<ARGS>(args)...); }
+  // syntactic sugar
+  template<typename... ARGS>
+  void detail(ARGS&&... args) { log<TextDetail>(std::forward<ARGS>(args)...); }
 
-    template<typename... ARGS>
-    void info(ARGS&&... args) { log<TextInfo>(std::forward<ARGS>(args)...); }
+  template<typename... ARGS>
+  void info(ARGS&&... args) { log<TextInfo>(std::forward<ARGS>(args)...); }
 
-    template<typename... ARGS>
-    void warn(ARGS&&... args) { log<TextWarning>(std::forward<ARGS>(args)...); }
+  template<typename... ARGS>
+  void warn(ARGS&&... args) { log<TextWarning>(std::forward<ARGS>(args)...); }
 
-    template<typename... ARGS>
-    void error(ARGS&&... args) { log<TextError>(std::forward<ARGS>(args)...); }
+  template<typename... ARGS>
+  void error(ARGS&&... args) { log<TextError>(std::forward<ARGS>(args)...); }
 
-    template<typename... ARGS>
-    void csv(ARGS&&... args) { log<TextCSV>(std::forward<ARGS>(args)...); }
+  template<typename... ARGS>
+  void csv(ARGS&&... args) { log<TextCSV>(std::forward<ARGS>(args)...); }
 
-  protected:
-    char const* name;
-  };
+protected:
+  char const* name;
+};
 
-  template<class Filter>
-  template<class MSG, typename... ARGS>
-  void Logger<Filter>::write(ARGS&&... args) const {
-    MSG msg(name, std::forward<ARGS>(args)...);
-    sink->write(msg.getData(), msg.getSize());
+template<class Filter>
+template<class MSG, typename... ARGS>
+void Logger<Filter>::write(ARGS&&... args) const {
+  MSG msg(name, std::forward<ARGS>(args)...);
+  sink->write(msg.getData(), msg.getSize());
+}
+
+static Logger<FilterAny> testLog("Test");
+inline static void test_log(const char *file_line, const char *msg, const char *expr) {
+  if (testLog.isActive(mlog::TextError::VERBOSITY)) {
+    testLog.write<mlog::TextError>(file_line, msg, expr);
   }
+}
+
+#define MLOG_TEST_OP(logger, op, val, expected) \
```

Comment:
Test results should be logged always independent of logging verbosity and test failure or success. Otherwise, it is difficult to prove that the tests were actually done. You could add two message types in TextMsg.hh for failure and success. And then use the Logger::write without going through MLOG_ERROR.